### PR TITLE
Allow replay requests to disable Replay optimization.

### PR DESCRIPTION
**Pull request by AWoloszyn**

### CODE REVIEW

**Comment by Qining** [comment]
lgtm

File: cmd/gapit/screenshot.go [hunk 1]
```diff
@@ -97,7 +97,12 @@ func (verb *screenshotVerb) writeSingleFrame(frame image.Image, fn string) error
 func getSingleFrame(ctx context.Context, cmd *path.Command, device *path.Device, client service.Service) (*image.NRGBA, error) {
 	ctx = log.V{"cmd": cmd.Indices}.Bind(ctx)
 	settings := &service.RenderSettings{MaxWidth: uint32(0xFFFFFFFF), MaxHeight: uint32(0xFFFFFFFF)}
-	iip, err := client.GetFramebufferAttachment(ctx, device, cmd, api.FramebufferAttachment_Color0, settings, nil)
+	iip, err := client.GetFramebufferAttachment(ctx,
+		&service.ReplaySettings{
+			Device: device,
+			DisableReplayOptimization: false,
```
Comment (ben-clayton) [inline cmd/gapit/screenshot.go]: Nit: you can simply omit default value fields.

File: cmd/gapit/video.go [hunk 2]
```diff
@@ -324,7 +324,10 @@ func (verb *videoVerb) encodeVideo(ctx context.Context, filepath string, vidFun
 func getFrame(ctx context.Context, maxWidth, maxHeight int, cmd *path.Command, device *path.Device, client service.Service) (*image.NRGBA, error) {
 	ctx = log.V{"cmd": cmd.Indices}.Bind(ctx)
 	settings := &service.RenderSettings{MaxWidth: uint32(maxWidth), MaxHeight: uint32(maxHeight)}
-	iip, err := client.GetFramebufferAttachment(ctx, device, cmd, api.FramebufferAttachment_Color0, settings, nil)
+	iip, err := client.GetFramebufferAttachment(ctx, &service.ReplaySettings{
+		Device: device,
+		DisableReplayOptimization: false,
```
Comment (ben-clayton) [inline cmd/gapit/video.go]: Ditto

File: gapis/resolve/thumbnail.go [hunk 3]
```diff
@@ -41,9 +41,12 @@ func Thumbnail(ctx context.Context, p *path.Thumbnail) (*image.Info, error) {
 }
 
 // CommandThumbnail resolves and returns the thumbnail for the framebuffer at p.
-func CommandThumbnail(ctx context.Context, w, h uint32, f *image.Format, p *path.Command) (*image.Info, error) {
+func CommandThumbnail(ctx context.Context, w, h uint32, f *image.Format, noOpt bool, p *path.Command) (*image.Info, error) {
```
Comment (ben-clayton) [inline gapis/resolve/thumbnail.go]: Maybe standardize all these APIs with `ReplaySettings`?

File: gapis/service/path/path.proto [hunk 4]
```diff
@@ -428,4 +428,6 @@ message Thumbnail {
         Command command = 5;
         CommandTreeNode command_tree_node = 6;
     }
+
+    bool disableOptimization = 7;
```
Comment (ben-clayton) [inline gapis/service/path/path.proto]: Maybe use ReplaySettings here too? Might require moving it to this package though.

File: cmd/gapit/screenshot.go (hunk 1 above)
Comment (AWoloszyn) [inline cmd/gapit/screenshot.go]: Done

File: cmd/gapit/video.go (hunk 2 above)
Comment (AWoloszyn) [reply cmd/gapit/video.go]: Done

**Pull request merged**